src/ddprof_worker.cc

```diff
@@ -412,7 +421,7 @@ DDRes ddprof_pr_sysallocation_tracking(DDProfContext *ctx,
   } else if (id == 60 || id == 231 || id == 59 || id == 322 || id == 520 ||
              id == 545) {
     // Erase upon exit or exec
-    sysalloc.do_exit(sample->pid);
+    clear_pid = true;
```

Comment:
All the syscalls here seem hardcoded for x86_64 / x86.
Do we really want to extract pid lifetime information from the syscalls (there are already perf events + process alive check) ?

Reply:
I agree this is probably redundant

Reply:
1. yeah, these are hardcoded x86-specific syscalls.  I have a patch in a different branch that makes this better, although it doesn't add arm support...

2. I agree that the global PID lifetime probably shouldn't be inferred in pieces all over the application.  If an individual profiling type, which is respons